decam_process.py

```diff
@@ -15,324 +15,378 @@
-$ python hits_ingest.py processCcd
-$ python hits_ingest.py diffIm
+$ python decam_process.py -d dataset_root -o output_location -i "visit=12345, ccd=5"
+
+NOTE: -i flag is not implemented yet. For now, set the visit and ccd in main().
 
-A typical workflow will run these four tasks in order. The user must set
-repo, calibrepo, processedrepo, diffimrepo, visits, and ccdnum in the code.
+TODO: Update DMTN-039 to reflect the new user interface.
+      I'm postponing this until DM-11422 and/or DM-11390 are complete.
 '''
 
 
 def main():
     '''
-    Set input parameters (repos, visits, ccdnums), parse command-line args,
-    and run the requested task.
-    '''
-
-    # ~~  edit values below as desired  ~~ #
-    repo = 'ingested_15A38/'  # used by ingest, ingestCalibs, processCcd
-    calibrepo = 'calibingested_15A38/'  # used by ingestCalibs, processCcd
-    processedrepo = 'processed_15A38/'  # used by processCcd, diffIm
-    diffimrepo = 'diffim_15A38_g/'  # used by diffIm
-    visits = [410927, 411033]  # used by processCcd, diffIm
-    # NOTE: visits assumes the first element is template and the rest are science
-    ccdnum = '1..62'  # used by processCcd, diffIm
-    # NOTE: the default '1..62' value includes all of the DECam CCDs
-    # ~~ edit values above as desired  ~~ #
-
-    # Parse visits into strings for command-line arguments
-    visit = '^'.join(str(v) for v in visits)
-    templatevisit = str(visits[0])
-    sciencevisit = '^'.join(str(v) for v in visits[1:])
+    Parse command-line args and run the pipeline. NOT used by ap_verify.
 
+    This main function is solely for the purpose of running ap_pipe alone,
+    from the command line, on a dataset intended for ap_verify. It is useful
+    for testing or standalone image processing independently from verification.
+    '''
     # Parse command line arguments with argparse
     parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
                                      description=textwrap.dedent('''
     Process raw decam images with MasterCals from ingestion --> difference imaging
 
     USAGE:
-    $ python hits_ingest.py ingest -f path/to/rawimages/
-    $ python hits_ingest.py ingestCalibs -f path/to/calibrations/somefiles*.fits.fz
-    $ python hits_ingest.py processCcd
-    $ python hits_ingest.py diffIm
-
-    A typical workflow will run these four tasks in order. The user must set
-    repo, calibrepo, processedrepo, diffimrepo, visits, and ccdnum in the code.
+    $ python decam_process.py -d dataset_root -o output_location -i "visit=12345, ccd=5"
                                      '''))
-    parser.add_argument('task', choices=['ingest', 'ingestCalibs', 'processCcd', 'diffIm'],
-                        help=textwrap.dedent('''
-    Which of four tasks you would like to run.
-
-    ingest - Ingest raw DECam images into a repository with a
-    corresponding registry.
-    USAGE:
-    $ python hits_ingest.py ingest -f path/to/rawimages/
```

Comment:
Essentially yes; a call to run the pipeline script via `main()` will run all of the steps rather than just one at a time. This doesn't matter for `ap_verify` (or any other external code using the pipeline) because it can call whichever function it wants whenever it wants. Is that OK?